docs/vagrant-spk/debugging.md

```diff
@@ -0,0 +1,245 @@
+consider the following.
+
+- `/` (read-only) contains all the files that could hypothetically go into your package. The bash process
+  launched by `vagrant-spk enter-grain`shell starts in `/`.
+
+- `/var` (read-write) contains the writable state for your app. This maps to
+  `/opt/sandstorm/var/sandstorm/grains/{{grainId}}/sandbox` outside the grain.
+
+If your `.sandstorm/sandstorm-pkgdef.capnp` contains a `sourcePath = "/"` line, which is the default
+for all vagrant-spk platform stacks except the Meteor stack, then the full contents of the Vagrant
+base box are available. If not, read below on how to add that.
+
+`vagrant-spk` configures `.sandstorm/Vagrantfile` so that `/opt/app` (read-only within the grain) is
+a shared folder to your host operating system, typically containing your app's source code. If your
+app contains command-line tools, then you probably need to run this command before you can access
+those tools.
+
+```bash
+cd /opt/app
+```
+
+### Overcoming missing command-line tools like ls
+
+Some apps, especially those using Meteor platform stack for vagrant-spk, do not configure `sourcePath = "/"`,
+so if you try to run some commands, you will get an error. For example:
+
+```bash
+$ ls
+bash: ls: command not found
+```
+
+To work around this, you have three options.
+
+**Option 1. Use debugging tools that are available.** For the Meteor stack in particular, you can
+choose to run `node` instead, which provides a more Meteor-like experience while debugging. Here is
+how you can launch node from `vagrant-spk enter-grain`.
+
+```bash
+$ vagrant-spk enter-grain
+$ /bin/node
+>   # you can now run whatever Javascript you wish
+```
+
+**Option 2. Inject the specific tools you need.** One good way to do this is to inject busybox into
+the grain. To do that, you can run the following.
+
+```bash
+$ vagrant-spk vm ssh
+$ sudo cp /bin/busybox /opt/sandstorm/var/sandstorm/grains/{{grainId}}/sandbox
+```
+
+Now, when you enter the grain, you can use busybox's bundled versions of ls, cp, and other typical
+tools.
+
+```bash
+$ vagrant-spk enter-grain
+$ ls
+bash: ls: command not found
+bash-4.3$ /var/busybox sh
+$ ls  # will work now, since it comes from busybox
+```
+
+**Option 3. Enable filesystem tracing for this app.** In `.sandstorm/sandstorm-pkgdef.capnp`, look for this
+line, which disables tracing.
+
+```bash
+alwaysInclude = [ "." ] ,
+```
+
+Replace it with this line, which enables tracing.
+
+```bash
+alwaysInclude = [ ],
+```
+
+Additionally, look for the line starting with `sourceMap = `. Within that section, make sure that this
+line is present. If not, you should add it.
+
+```bash
+      ( sourcePath = "/" ),
+```
+
+Finally, look for the terminal window where you are running `vagrant-spk dev`. Use Ctrl-C to terminate
+it, then run it again.
+
+This will dramatically bloat your sandstorm-files.list, so I recommend you **undo this change before
```

Comment:
Do we use "I" or "we" in the docs?  I think I've seen both; we should probably attempt to be consistent.  (I prefer "we" in documentation maintained by multiple individuals, unless there's also clearly attributed authorship for each article/unit, so we know who is "I".)